parsec/core/mountpoint/fuse_runner.py

```diff
@@ -232,13 +232,26 @@ async def _stop_fuse_thread(
 
     # Fuse exit in macOS
     if sys.platform == "darwin":
+
         # The schedule_exit() solution doesn't work on macOS, instead freezes the application for
         # 120 seconds before a timeout occurs. The solution used is to call this function (macOS
         # equivalent to fusermount) in a subprocess to unmount.
-        await trio.run_process(["diskutil", "unmount", "force", str(mountpoint_path)])
+        process_args = ["diskutil", "unmount", "force", str(mountpoint_path)]
+        process = await trio.open_process(process_args)
```

Comment:
What's the point of using `open_process` instead of `run_process` here ?

Reply:
it's basically the same logic as the one with the daemon thread below, we fire a background task that is supposed to cause the FS thread to stop, and we wait for the corresponding event while restarting the background task if necessary.

Reply:
See this code within the loop:
```python
            if process.poll() is not None:
                process = await trio.open_process(process_args)
```